core/src/main/java/org/keycloak/TokenVerifier.java

```diff
@@ -366,11 +366,18 @@ private void removeCheck(Predicate<? super T> check) {
     /**
      * Add check for verifying that token contains the expectedAudience
      *
-     * @param expectedAudience Audience, which needs to be in the target token. Can't be null
+     * @param expectedAudiences Audiences, which needs to be in the target token. Can be <code>null</code>.
      * @return This token verifier
      */
-    public TokenVerifier<T> audience(String expectedAudience) {
-        return this.replaceCheck(AudienceCheck.class, true, new AudienceCheck(expectedAudience));
+    public TokenVerifier<T> audience(String... expectedAudiences) {
+        if (expectedAudiences == null || expectedAudiences.length == 0) {
+            return this.replaceCheck(AudienceCheck.class, true, new AudienceCheck(null));
```

Comment:
Just a silly question. For some reason, originally it was prohibited for `expectedAudience` to be null. Why to allow it now? And are we 100% sure this doesn't create e.g. some vulnerability?

Reply:
The behavior remains the same here. In the old code, (with `String expectedAudience`), if you pass null there, you will effectively invoke `return this.replaceCheck(AudienceCheck.class, true, new AudienceCheck(null));`

The problem with the updated code is that the argument can be a null or an empty array. In both cases, I'm explicitly adding `return this.replaceCheck(AudienceCheck.class, true, new AudienceCheck(null));`.

In both cases, this blows up in the runtime [here](https://github.com/keycloak/keycloak/blob/39b05b499c238899bb9d02ef681e500dc58fd38e/core/src/main/java/org/keycloak/TokenVerifier.java#L148).